perses/rjmc/topology_proposal.py

```diff
@@ -2366,18 +2374,23 @@ def _constraint_repairs(atom_map, old_system, new_system, old_topology, new_topo
             elif atom2 in new_hydrogens:
                 new_constraints[atom2] = length
 
-         #iterate through the atom indices in the new_to_old map, check bonds for pairs, and remove appropriate matches
+        # iterate through the atom indices in the new_to_old map, check bonds for pairs, and remove appropriate matches
         to_delete = []
         for new_index, old_index in atom_map.items():
-            if new_index in new_constraints.keys() and old_index in old_constraints.keys(): # both atom indices are hydrogens
+            if new_index in new_constraints.keys() and old_index in old_constraints.keys():  # both atoms in constraints
                 old_length, new_length = old_constraints[old_index], new_constraints[new_index]
-                if not old_length == new_length: #then we have to remove it from
+                if not old_length == new_length:  # then we have to remove it from
                     to_delete.append(new_index)
+            # demap atoms that were constrained and now are in an unconstrained edge, and vice versa
+            elif old_index in old_constraints.keys() and new_index not in new_constraints.keys():
+                to_delete.append(new_index)
+            elif old_index not in old_constraints.keys() and new_index in new_constraints.keys():
+                to_delete.append(new_index)
 
         for idx in to_delete:
             del atom_map[idx]
 
-        return atom_map
+        return atom_map, to_delete
```

Comment:
It may not even be necessary to return the `to_delete` if we can just take the difference in atom maps.

Reply:
I guess we could convert them to sets and get the difference, new - old, but I'm not 100% sure this is safe in the long term, especially if we want to use `AtomMapping` objects instead of dictionaries or sets.